samcli/lib/pipeline/bootstrap/stage_resources.yaml

```diff
@@ -94,31 +120,55 @@ Resources:
         - Key: Role
           Value: pipeline-execution-role
       AssumeRolePolicyDocument:
-        Version: 2012-10-17
-        Statement:
-          - Effect: Allow
-            Principal:
-              AWS:
-                - Fn::If:
-                  - MissingPipelineUser
-                  - !GetAtt PipelineUser.Arn
-                  - !Ref PipelineUserArn
-            Action:
-              - 'sts:AssumeRole'
-          - Effect: Allow
-            Principal:
-              # Allow roles with tag Role=aws-sam-pipeline-codebuild-service-role to assume this role.
-              # This is required when CodePipeline is the CI/CD system of choice.
-              AWS:
-                - !If
-                  - MissingPipelineUser
-                  - !Ref AWS::AccountId
-                  - !Select [4, !Split [':', !Ref PipelineUserArn]]
-            Action:
-              - 'sts:AssumeRole'
-            Condition:
-              StringEquals:
-                aws:PrincipalTag/Role: aws-sam-pipeline-codebuild-service-role
+        Fn::If:
+          - DontUseOidc
+          - Version: 2012-10-17
+            Statement:
+              - Effect: Allow
+                Principal:
+                  AWS:
+                    - Fn::If:
+                      - MissingPipelineUser
+                      - !GetAtt PipelineUser.Arn
+                      - !Ref PipelineUserArn
+                Action:
+                  - 'sts:AssumeRole'
+              - Effect: Allow
+                Principal:
+                  # Allow roles with tag Role=aws-sam-pipeline-codebuild-service-role to assume this role.
+                  # This is required when CodePipeline is the CI/CD system of choice.
+                  AWS:
+                    - !If
+                      - MissingPipelineUser
+                      - !Ref AWS::AccountId
+                      - !Select [4, !Split [':', !Ref PipelineUserArn]]
+                Action:
+                  - 'sts:AssumeRole'
+                Condition:
+                  StringEquals:
+                    aws:PrincipalTag/Role: aws-sam-pipeline-codebuild-service-role
+          - Fn::Sub: 
+              - |
+                {
+                  "Version": "2012-10-17",
+                  "Statement": [
+                    {
+                      "Effect": "Allow",
+                      "Principal": {
+                        "Federated": "arn:aws:iam::${AWS::AccountId}:oidc-provider/${Url}"
```

Comment:
use AWS::Partition instead of :aws: , other places need to be cross checked as well.

Reply:
Good catch! Will update it.